src/test_typing_extensions.py

```diff
@@ -2210,6 +2218,36 @@ def test_or_and_ror(self):
             Union[typing_extensions.Generator, typing.Deque]
         )
 
+    def test_setattr(self):
+        origin = collections.abc.Generator
+        alias = typing_extensions.Generator
+
+        # Attribute assignment on generic alias sets attribute on origin
```

Comment:
You could possibly do something like this? The advantage of using `addCleanUp` is you know the cleanup function will be called even if the test method is interrupted halfway through the test due to one of the test's assertions failing

```suggestion
        alias = typing_extensions.Generator
        original_name = alias._name
        
        def cleanup():
            for obj in origin, alias:
                for attr in 'foo', '__dunder__':
                    try:
                        delattr(obj, attr)
                    except Exception:
                        pass

            try:
                alias._name = original_name
            except Exception:
                pass

        self.addCleanup(cleanup)

        # Attribute assignment on generic alias sets attribute on origin
```